include/vinecopulib/misc/implementation/tools_integration.ipp

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2016-2019 Thomas Nagler and Thibault Vatter
+//
+// This file is part of the vinecopulib library and licensed under the terms of
+// the MIT license. For a copy, see the LICENSE file in the root directory of
+// vinecopulib or https://vinecopulib.github.io/vinecopulib/.
+
+#include <iostream>
+#include <vinecopulib/misc/tools_integration_legendre.hpp>
+
+namespace vinecopulib {
+
+//! Utilities for integration
+namespace tools_integration {
+
+//! @brief load quadrature nodes and weights for numerical integration
+//!
+//! @param n number of nodes and weights (10, 20, 30, 40 or 50)
+//!
+//! @return An \f$ n \times 2 \f$ with the nodes in the first colum and the
+//! weights in the second.
+inline Eigen::Matrix<double, Eigen::Dynamic, 2>
+quadrature_rule(const size_t n)
+{
+  Eigen::Matrix<double, Eigen::Dynamic, 2> output;
+  switch (n) {
+    case 10:
+      output = tools_legendre::nodes_weights_10;
+      break;
+    case 20:
+      output = tools_legendre::nodes_weights_20;
+      break;
+    case 30:
+      output = tools_legendre::nodes_weights_30;
+      break;
+    case 40:
+      output = tools_legendre::nodes_weights_40;
+      break;
+    case 50:
+      output = tools_legendre::nodes_weights_50;
+      break;
+    default:
```

Comment:
I assume that we will always just use the default `n` (currently 20). I'd vote to keep the code base lean by removing the option altogether, possibly changing the default to something a bit larger. Did you do some accuracy checks?

Reply:
Started a preliminary implementation of conditional distributions. Seems like we need **at least** `n = 50`. Relative error was still around 5-10% :(

Reply:
Yeah, that was at that point that I stopped: I could not get the errors at a acceptably low level :(